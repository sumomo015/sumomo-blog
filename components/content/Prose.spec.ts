// @vitest-environment nuxt

import { describe, it, expect } from 'vitest'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import { ProseA, ProseCodeInline, ProseH1, ProseH2, ProseH3, ProseLi, ProseP, ProseUl } from '#components'

describe('Prose', () => {
  it('ProseA', async () => {
    const wrapper = await mountSuspended(ProseA, {
      props: { href: 'https://example.com' },
      slots: { default: 'リンク' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseCodeInline', async () => {
    const wrapper = await mountSuspended(ProseCodeInline, {
      slots: { default: 'インラインコード' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseH1', async () => {
    const wrapper = await mountSuspended(ProseH1, {
      slots: { default: '見出し1' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseH2', async () => {
    const wrapper = await mountSuspended(ProseH2, {
      slots: { default: '見出し2' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseH3', async () => {
    const wrapper = await mountSuspended(ProseH3, {
      slots: { default: '見出し3' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseLi', async () => {
    const wrapper = await mountSuspended(ProseLi, {
      slots: { default: 'リストアイテム' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseP', async () => {
    const wrapper = await mountSuspended(ProseP, {
      slots: { default: '段落' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('ProseUl', async () => {
    const wrapper = await mountSuspended(ProseUl, {
      slots: { default: 'リスト' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
