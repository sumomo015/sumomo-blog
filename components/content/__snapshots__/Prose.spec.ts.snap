// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Prose > ProseA 1`] = `"<a href="https://example.com" rel="noopener noreferrer" target="_blank" class="font-medium base-color-link hover:underline">リンク</a>"`;

exports[`Prose > ProseCodeInline 1`] = `"<code class="px-1.5 py-1 rounded-md text-sm font-semibold base-color-header border-1 border-gray-200 dark:border-gray-700 bg-gray-100 dark:bg-gray-800">インラインコード</code>"`;

exports[`Prose > ProseH1 1`] = `"<h1 class="mb-6 mt-12 scroll-mt-20 text-3xl font-bold base-color-header">見出し1</h1>"`;

exports[`Prose > ProseH2 1`] = `"<h2 class="mb-6 mt-12 scroll-mt-20 text-2xl font-bold base-color-header">見出し2</h2>"`;

exports[`Prose > ProseH3 1`] = `"<h2 class="mb-5 mt-8 scroll-mt-20 text-xl font-bold base-color-header">見出し3</h2>"`;

exports[`Prose > ProseLi 1`] = `"<li class="my-1 ps-1.5">リストアイテム</li>"`;

exports[`Prose > ProseP 1`] = `"<p class="my-5">段落</p>"`;

exports[`Prose > ProseUl 1`] = `"<ul class="my-2 list-disc list-outside ps-6">リスト</ul>"`;
